pipeline-model-definition/src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -796,16 +796,21 @@ public class Utils {
         Set<String> previousTriggers = new HashSet<>()
         Set<String> previousParameters = new HashSet<>()
 
-        DeclarativeJobPropertyTrackerAction previousAction = null
-        WorkflowRun previousBuild = r.getPreviousBuild()
-        if (previousBuild != null) {
-            previousAction = previousBuild.getAction(DeclarativeJobPropertyTrackerAction.class)
-            if (previousAction != null) {
-                previousProperties.addAll(previousAction.getJobProperties())
-                previousTriggers.addAll(previousAction.getTriggers())
-                previousParameters.addAll(previousAction.getParameters())
+        // First, use the action from the job if it's present.
+        DeclarativeJobPropertyTrackerAction previousAction = j.getAction(DeclarativeJobPropertyTrackerAction.class)
```

Comment:
(FTR this is why I recommended waiting to develop the `workflow-multibranch` fix before pushing anything out—to avoid needing to think about multiple upgrade scenarios.)